### PR TITLE
Feature/native camera

### LINKED ISSUE #485: Support for basic camera control
Zoom in/out to start with would be good.

### LINKED ISSUE #505: Support for iOS native camera

### PR DESCRIPTION
**Pull request by sklencar**
Added native camera support for android and ios.
Dependent on QGIS PR: https://github.com/qgis/QGIS/pull/33995

closes #505 and closes #485

### CODE REVIEW

**Comment by PeterPetrik** [comment]
@sklencar can you please rebase based on the current master? please do not modify any gradle and xml files

File: app/android/build.gradle [hunk 1]
```diff
@@ -4,6 +4,7 @@ buildscript {
     repositories {
         google()
         jcenter()
+        google()
```
Comment (PeterPetrik) [inline app/android/build.gradle:7]: Do not edit this file

File: app/android.pri [hunk 2]
```diff
@@ -7,7 +7,11 @@ android {
     # this makes the manifest visible from Qt Creator
     DISTFILES += $$OUT_PWD/android_patched/AndroidManifest.xml
     DISTFILES += $$OUT_PWD/res/xml/file_paths.xml
-    DISTFILES += $$PWD/build.gradle
```
Comment (PeterPetrik) [inline app/android.pri:10]: Do not edit this file

File: app/android/build.gradle (hunk 1 above)
Comment (PeterPetrik) [reply app/android/build.gradle:7]: Do not add app/android/gradle/wrapper/gradle-wrapper.jar file

File: app/android/gradle/wrapper/gradle-wrapper.properties [hunk 3]
```diff
@@ -0,0 +1,6 @@
+#Mon Feb 20 10:43:22 EST 2017
```
Comment (PeterPetrik) [inline app/android/gradle/wrapper/gradle-wrapper.properties:1]: do not add new files

File: app/android/gradlew [hunk 4]
```diff
@@ -0,0 +1,172 @@
+#!/usr/bin/env sh
```
Comment (PeterPetrik) [inline app/android/gradlew:1]: Do not add new files!

File: app/android/gradlew.bat [hunk 5]
```diff
@@ -0,0 +1,84 @@
+@if "%DEBUG%" == "" @echo off
```
Comment (PeterPetrik) [inline app/android/gradlew.bat:1]: Do not add new files!

File: app/android/gradle/wrapper/gradle-wrapper.properties [hunk 6]
```diff
@@ -0,0 +1,5 @@
+distributionBase=GRADLE_USER_HOME
```
Comment (PeterPetrik) [inline app/android/gradle/wrapper/gradle-wrapper.properties:1]: remove!

**Comment by PeterPetrik** [comment]
@sklencar is this WIP: or finished? is something missing?

**Comment by saberraz** [comment]
Testing on Android 9:
It presents the camera with all the settings...photo is taken too, but it does not save the photo or even the feature.

**Comment by saberraz** [comment]
> Testing on Android 9:
> It presents the camera with all the settings...photo is taken too, but it does not save the photo or even the feature.

This has been resolved now with sdk-12

**Pull request merged**